protocol/statetracker/spec_updater.go

```diff
@@ -0,0 +1,65 @@
+package statetracker
+
+import (
+	"context"
+
+	"github.com/lavanet/lava/protocol/chainlib"
+	"github.com/lavanet/lava/utils"
+	spectypes "github.com/lavanet/lava/x/spec/types"
+)
+
+const (
+	CallbackKeyForSpecUpdate = "spec-update"
+)
+
+type SpecGetter interface {
+	GetSpec(ctx context.Context, chainID string) (*spectypes.Spec, error)
+}
+
+type SpecUpdater struct {
+	eventTracker     *EventTracker
+	chainParser      chainlib.ChainParser
```

Comment:
remove chainParser reference, use an interface (you only need SetSpec i think), interface can be called SpecUpdatable, and then we can call it once for all api interfaces, the key and arguments shouldn't get apiInterface